specs/vep-2272-complete-data-job-configuration-persistence/README.md

```diff
@@ -0,0 +1,161 @@
+
+## Motivation
+
+Ensuring efficient and reliable management of data job deployments is crucial for the smooth functioning of our system.
+Currently, when a data job is deployed, its deployment configuration is stored in both Kubernetes and the database.
+However, there exists a disparity where certain essential properties are exclusively stored in Kubernetes, such as the
+job's enabled status, Python version, and deployment information.
+
+This discrepancy creates a significant challenge for the Control Service APIs that rely on retrieving data job
+deployments. For instance, the GraphQL API's jobs query initially retrieves the data jobs from the database and
+subsequently retrieves the deployment data from Kubernetes. This two-step process introduces performance degradation to
+the Control Service, as it requires additional time and resources to fetch information from multiple sources.
+
+Moreover, in the event of a Kubernetes namespace loss, there is a potential risk of losing critical data job deployment
+configurations. In such scenarios, recovering the lost data jobs becomes a manual and complex restoration process.
+This adds further complexity and potential downtime for users relying on the Control Service for their data job
+management needs.
+
+To address these issues, it is essential to optimize the storage and retrieval of data job deployments. By ensuring that
+all relevant properties are stored consistently, in the database, so we can streamline the process and enhance the
+overall the efficiency of the Control Service APIs.
+
+By improving the efficiency and reliability of data job deployments, we aim to optimize the overall system performance
+and provide a seamless experience for our users.
+
+## Requirements and goals
+
+### Goals
+
+1. Identify the specific data job configuration that is stored only within the Kubernetes cluster.
+2. Transition the primary source of truth for all data jobs configurations from both Kubernetes and database to just the
+   database, in order to centralize data storage, improve data integrity, and enable easier access and manipulation.
+3. Enhance the performance of Control Service APIs by loading the data job configuration from only one data source
+   (Control Service database).
+4. Prepare Control Service for future enhancement of the migration process and ease of restoration by centralizing and
+   storing all data job configurations within the database.
+
+### Non-Goals
+
+1. Implement automatic data jobs restore capability in case of the loss of Kubernetes namespace whereby the Control
+   Service automatically retrieves the configuration from a database and seamlessly restores the data jobs.
+
+## High-level design
+
+These diagrams below show a high-level design view of the changes that would be proposed in this VEP.
+
+|                       Deploy Data Job                       |                           Read Data Job Deployment Configurations                           |
```

Comment:
I believe "Deploy Data Job" should be "Deploy or Update Data Job" as updates to the data job configuration will also be persisted in the DB, right?

Reply:
Correct